src/status_im/contexts/wallet/common/token_value/view.cljs

```diff
@@ -3,50 +3,63 @@
-         :on-press            #(js/alert "to be implemented")}
-        {:icon                :i/bridge
-         :accessibility-label :bridge
-         :label               (i18n/label :t/bridge)
-         :on-press            #(js/alert "to be implemented")}
-        {:icon                :i/settings
-         :accessibility-label :settings
-         :label               (i18n/label :t/manage-tokens)
-         :on-press            #(js/alert "to be implemented")
-         :add-divider?        true}
-        {:icon                :i/hide
-         :accessibility-label :hide
-         :label               (i18n/label :t/hide)
-         :on-press            #(js/alert "to be implemented")}]]]]))
+    [quo/action-drawer
+     [[{:icon                :i/buy
+        :accessibility-label :buy
+        :label               (i18n/label :t/buy)
+        :on-press            #(js/alert "to be implemented")
+        :right-icon          :i/external}
+       {:icon                :i/send
+        :accessibility-label :send
+        :label               (i18n/label :t/send)
+        :on-press            (fn []
+                               (rf/dispatch [:hide-bottom-sheet])
+                               (rf/dispatch [:wallet/clean-send-data])
+                               (rf/dispatch [:wallet/send-select-token
+                                             {:token       token-data
+                                              :start-flow? true}]))}
+       {:icon                :i/receive
+        :accessibility-label :receive
+        :label               (i18n/label :t/receive)
+        :on-press            #(js/alert "to be implemented")}
+       {:icon                :i/bridge
+        :accessibility-label :bridge
+        :label               (i18n/label :t/bridge)
+        :on-press            #(js/alert "to be implemented")}
+       {:icon                :i/settings
+        :accessibility-label :settings
+        :label               (i18n/label :t/manage-tokens)
+        :on-press            #(js/alert "to be implemented")
+        :add-divider?        true}
+       {:icon                :i/hide
+        :accessibility-label :hide
+        :label               (i18n/label :t/hide)
+        :on-press            #(js/alert "to be implemented")}]]]))
 
 (defn view
   [item _ _ {:keys [watch-only?]}]
   [quo/token-value
-   (cond-> item
-     (not watch-only?)
-     (assoc :on-long-press
-            #(rf/dispatch
-              [:show-bottom-sheet
-               {:content       (fn [] [token-value-drawer item])
-                :selected-item (fn [] [quo/token-value item])}])))])
+   (merge item
+          {:on-long-press
+           #(rf/dispatch
+             [:show-bottom-sheet
+              {:content       (fn []
+                                (if watch-only?
+                                  [watch-only-token-value-drawer]
+                                  [token-value-drawer item]))
```

Comment:
I would prefer we just have one token-value-drawer component and we filter/ configure the options based on certain props. That's how we do it in other parts of the codebase

Reply:
If you look up other uses of token-drawer you'll see what I mean.

The thing is that both of these watch only / regular account will likely share the same options and one is a subset of the other